ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,148 @@
+from functools import cached_property
+from ansible_collections.arista.avd.plugins.module_utils.utils import get
+
+
+class AVDStructConfig(EosDesigns):
+    def __init__(
+            self,
+            host_vars=None,
+            combine=None,
+            list_compress=None,
+            range_expand=None,
+            convert_dicts=None,
+            natural_sort=None,
+    ):
+        if host_vars:
+            self._host_vars = host_vars
+        if combine:
+            self._combine = combine
+        if list_compress:
+            self._list_compress = list_compress
+        if range_expand:
+            self._range_expand = range_expand
+        if convert_dicts:
+            self._convert_dicts = convert_dicts
+        if natural_sort:
+            self._natural_sort = natural_sort
+        # super().__init__()
+
+    @cached_property
+    def _mgmt_interface_vrf(self):
+        return get(self._host_vars, "mgmt_interface_vrf")
+
+    @cached_property
+    def _mgmt_gateway(self):
+        return get(self._host_vars, "mgmt_gateway")
+
+    @cached_property
+    def _mgmt_destination_networks(self):
+        return get(self._host_vars, "mgmt_destination_network")
+
+    @cached_property
+    def _switch_data(self):
+        return get(self._host_vars, 'switch', {})
+
+    @cached_property
+    def _switch_bgp_as(self):
+        return get(self._switch_data,'bgp_as')
+
+    @cached_property
+    def _switch_router_id(self):
+        return get(self._switch_data, 'router_id')
+
+    @cached_property
+    def _switch_bgp_defaults(self):
+        return get(self._switch_data, 'bgp_defaults')
+
+    @cached_property
+    def _bgp_maximum_paths(self):
+        return get(self._host_vars, "bgp_maximum_paths")
+
+    @cached_property
+    def _bgp_ecmp(self):
+        return get(self._host_vars, "bgp_ecmp")
+
+    @cached_property
+    def _load_interval_default(self):
+        return get(self._host_vars, "load_interval_default")
+
+    @cached_property
+    def _hardware_counters(self):
+        return get(self._host_vars, "hardware_counters")
+
+    @cached_property
+    def _hardware_counters_features(self):
+        return get(self._hardware_counters, 'features')
+
+    @cached_property
+    def _internal_vlan_order(self):
+        return get(self._host_vars, 'internal_vlan_order')
+
+    @cached_property
+    def _allocation(self):
+        return get(self._internal_vlan_order, 'allocation')
+
+    @cached_property
+    def _range(self):
+        return get(self._internal_vlan_order, 'range')
+
+    @cached_property
+    def _beginning(self):
+        return get(self._range, 'beginning')
+
+    @cached_property
+    def _ending(self):
+        return get(self._range, 'ending')
+
+    @cached_property
+    def static_routes(self):
+        static_routes = []
+        if "mgmt_destination_networks" in self._host_vars and len(self._mgmt_destination_networks) > 0:
```

Comment:
It would be simpler to read by just setting `mgmt_destinations_networks = get(self._hostvars, "mgmt_destination_networks")` inside the `static_routes` function and then run `if mgmt_destination_networks` since it will be None if not set.
Same elsewhere. All the small internal properties will only slow things down, since they are written to the internal __dict__.